src/Microsoft.AspNetCore.Razor.LanguageServer/Program.cs

```diff
@@ -81,9 +83,15 @@ public static async Task MainAsync(string[] args)
                         services.AddSingleton<ProjectSnapshotManagerAccessor, DefaultProjectSnapshotManagerAccessor>();
                         services.AddSingleton<TagHelperFactsService, DefaultTagHelperFactsService>();
                         services.AddSingleton<VisualStudio.Editor.Razor.TagHelperCompletionService, VisualStudio.Editor.Razor.DefaultTagHelperCompletionService>();
-                        services.AddSingleton<TagHelperCompletionService, DefaultTagHelperCompletionService>();
                         services.AddSingleton<TagHelperDescriptionFactory, DefaultTagHelperDescriptionFactory>();
 
+                        // Completion
+                        services.AddSingleton<Completion.TagHelperCompletionService, Completion.DefaultTagHelperCompletionService>();
+                        services.AddSingleton<RazorCompletionItemProvider, DirectiveCompletionItemProvider>();
+                        services.AddSingleton<RazorCompletionItemProvider, DirectiveAttributeCompletionItemProvider>();
+                        services.AddSingleton<RazorCompletionItemProvider, DirectiveAttributeParameterCompletionItemProvider>();
+                        services.AddSingleton<RazorCompletionItemProvider, DirectiveAttributeTransitionCompletionItemProvider>();
```

Comment:
Any reason this logic can't be merged with `DirectiveAttributeCompletionItemProvider`?

Reply:
It might eventually but for now it wouldn't work as expected in VS because we can't intermingle `@...` with HTML completions.